src/lib.rs

```diff
@@ -1365,6 +1364,109 @@ impl DnsCache {
     }
 }
 
+pub trait AsAddr {
+    fn as_addr(&self) -> Result<HashSet<Ipv4Addr>>;
```

Comment:
To make it clear, can we name this method `as_ipv4_addr_set`? or anything to convey that meaning? 

So that, if this trait supports more address family or data structures in future, it will be easy to add. 

If we don't plan to support any new address or data structure in future for this trait, then I would suggest to name the trait itself to `AsIPv4AddrSet` or something like that.

Reply:
maybe `as_ipv4_addrs`?

Reply:
Sure, `as_ipv4_addrs` sounds good.  And I think it looks more consistent and concise if the trait name is also changed to `AsIpv4Addrs` .  

A trait with a single method feels right in this case (which is what you have here).  If we ever need to support IPv6, it's probably better to define a new trait instead of adding new methods into this trait.